src/main/java/org/gridsuite/study/server/service/ConsumerService.java

```diff
@@ -398,367 +238,287 @@ public Consumer<Message<String>> consumeCaseImportFailed() {
+                        networkModificationTreeService.updateSecurityAnalysisResultUuid(receiverObj.getNodeUuid(), resultUuid);
+                        updateStatusType = NotificationService.UPDATE_TYPE_SECURITY_ANALYSIS_STATUS;
+                        updateResultType = NotificationService.UPDATE_TYPE_SECURITY_ANALYSIS_RESULT;
+                    } break;
+                    case SENSITIVITY_ANALYSIS: {
+                        networkModificationTreeService.updateSensitivityAnalysisResultUuid(receiverObj.getNodeUuid(), resultUuid);
+                        updateStatusType = NotificationService.UPDATE_TYPE_SENSITIVITY_ANALYSIS_STATUS;
+                        updateResultType = NotificationService.UPDATE_TYPE_SENSITIVITY_ANALYSIS_RESULT;
+                    } break;
+                    case SHORT_CIRCUIT: {
+                        String busId = msg.getHeaders().get(HEADER_BUS_ID, String.class);
+                        ShortcircuitAnalysisType analysisType = (busId == null) ?
+                                ShortcircuitAnalysisType.ALL_BUSES :
+                                ShortcircuitAnalysisType.ONE_BUS;
+                        if (analysisType == ShortcircuitAnalysisType.ALL_BUSES) {
+                            networkModificationTreeService.updateShortCircuitAnalysisResultUuid(receiverObj.getNodeUuid(), resultUuid);
+                            updateStatusType = NotificationService.UPDATE_TYPE_SHORT_CIRCUIT_STATUS;
+                            updateResultType = NotificationService.UPDATE_TYPE_SHORT_CIRCUIT_RESULT;
+                        } else {
+                            networkModificationTreeService.updateOneBusShortCircuitAnalysisResultUuid(receiverObj.getNodeUuid(), resultUuid);
+                            updateStatusType = NotificationService.UPDATE_TYPE_ONE_BUS_SHORT_CIRCUIT_STATUS;
+                            updateResultType = NotificationService.UPDATE_TYPE_ONE_BUS_SHORT_CIRCUIT_RESULT;
+                        }
+
+                    } break;
+                    case VOLTAGE_INITIALIZATION: {
+                        networkModificationTreeService.updateVoltageInitResultUuid(receiverObj.getNodeUuid(), resultUuid);
+                        updateStatusType = NotificationService.UPDATE_TYPE_VOLTAGE_INIT_STATUS;
+                        updateResultType = NotificationService.UPDATE_TYPE_VOLTAGE_INIT_RESULT;
+                    } break;
+                    case DYNAMIC_SIMULATION: {
+                        networkModificationTreeService.updateDynamicSimulationResultUuid(receiverObj.getNodeUuid(), resultUuid);
+                        updateStatusType = NotificationService.UPDATE_TYPE_DYNAMIC_SIMULATION_STATUS;
+                        updateResultType = NotificationService.UPDATE_TYPE_DYNAMIC_SIMULATION_RESULT;
+                    } break;
                 }
+
+                UUID studyUuid = networkModificationTreeService.getStudyUuidForNodeId(receiverObj.getNodeUuid());
```

Comment:
We don't have the studyUuid in the message headers ? that's a pity to make a BDD query for this no ?
Arf, no, Calcultion services don't know about studyUuid ok then.

Reply:
Indeed. I don't really the whole flux and am only supposed to refactor, so for now, I can only assume that there is a good reason.